packages/headless/src/app/commerce-engine/commerce-engine.ts

```diff
@@ -113,24 +113,30 @@ export function buildCommerceEngine(
     reducers: commerceEngineReducers,
   };
 
-  const engine = buildEngine(augmentedOptions, thunkArguments);
+  const internalEngine = buildEngine(augmentedOptions, thunkArguments);
+  const {state: _, ...engine} = internalEngine;
 
   engine.dispatch(setContext(options.configuration.context));
 
   return {
     ...engine,
 
-    get state() {
-      return engine.state;
+    get [stateKey]() {
+      return internalEngine.state;
     },
+
+    get configuration() {
+      return internalEngine.state.configuration;
+    },
+
```

Comment:
This is interesting !
why
```
{
    [stateKey]: internalEngine.state,
    configuration: internalEngine.state.configuration
}
```
is not working here ?

Reply:
Because we need to return the current value of the state every time

E.g.,

```javascript
const test = { state: 'old' };

function stagnantStateGetter(arg) {
  return {
    state: arg.state
  };
};

const bad = stagnantStateGetter(test);

console.log(bad.state);
// prints 'old'

test.state = 'new'
console.log(bad.state)
// still prints 'old' :`(

function freshStateGetter(arg) {
  return {
    get state() {
      return arg.state;
    }
  };
}

const good = freshStateGetter(test);

console.log(good.state)
// prints 'new'

test.state = 'newest'
console.log(good.state)
// prints 'newest' :D
```

Reply:
I was the feeling this was something like this. Thanks for the very detailled explanation. :bowing_man: Its very appreciated !